File: frameworks/rails/adsmedia_controller.rb
# ADSMedia Controller for Rails
# Add routes in config/routes.rb:
# namespace :api do
#   resources :emails, only: [] do
#     collection do
#       post :send_email
#       post :send_batch
#       get :check_suppression
#       get :ping
#     end
#   end
# end

class Api::EmailsController < ApplicationController
  skip_before_action :verify_authenticity_token

  def send_email
    result = adsmedia.send_email(
      to: params[:to],
      subject: params[:subject],
      html: params[:html],
      to_name: params[:to_name],
      from_name: params[:from_name],
      text: params[:text]
    )
    render json: { success: true, data: result }
  rescue StandardError => e
    render json: { error: e.message }, status: :internal_server_error
  end

  def send_batch
    result = adsmedia.send_batch(
      recipients: params[:recipients],
      subject: params[:subject],
      html: params[:html],
      text: params[:text],
      preheader: params[:preheader],
      from_name: params[:from_name]
    )
    render json: { success: true, data: result }
  rescue StandardError => e
    render json: { error: e.message }, status: :internal_server_error
  end

  def check_suppression
    result = adsmedia.check_suppression(params[:email])
    render json: { success: true, data: result }
  rescue StandardError => e
    render json: { error: e.message }, status: :internal_server_error
  end

  def ping
    result = adsmedia.ping
    render json: { success: true, data: result }
  rescue StandardError => e
    render json: { error: e.message }, status: :internal_server_error
  end

  private

  def adsmedia
    @adsmedia ||= ADSMedia.client
  end
end

File: frameworks/laravel/ADSMediaController.php
<?php

namespace App\Http\Controllers;

use App\Services\ADSMediaService;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

/**
 * Laravel Controller for ADSMedia
 */
class ADSMediaController extends Controller
{
    private ADSMediaService $adsmedia;

    public function __construct(ADSMediaService $adsmedia)
    {
        $this->adsmedia = $adsmedia;
    }

    public function send(Request $request): JsonResponse
    {
        $request->validate([
            'to' => 'required|email',
            'subject' => 'required|string',
            'html' => 'required|string',
            'to_name' => 'nullable|string',
            'from_name' => 'nullable|string',
            'text' => 'nullable|string',
        ]);

        try {
            $result = $this->adsmedia->send(
                $request->input('to'),
                $request->input('subject'),
                $request->input('html'),
                $request->input('to_name'),
                $request->input('from_name'),
                $request->input('text')
            );

            return response()->json(['success' => true, 'data' => $result]);
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }

    public function sendBatch(Request $request): JsonResponse
    {
        $request->validate([
            'recipients' => 'required|array',
            'recipients.*.email' => 'required|email',
            'subject' => 'required|string',
            'html' => 'required|string',
        ]);

        try {
            $result = $this->adsmedia->sendBatch(
                $request->input('recipients'),
                $request->input('subject'),
                $request->input('html'),
                $request->input('text'),
                $request->input('preheader'),
                $request->input('from_name')
            );

            return response()->json(['success' => true, 'data' => $result]);
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }

    public function checkSuppression(Request $request): JsonResponse
    {
        $request->validate(['email' => 'required|email']);

        try {
            $result = $this->adsmedia->checkSuppression($request->query('email'));
            return response()->json(['success' => true, 'data' => $result]);
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }

    public function ping(): JsonResponse
    {
        try {
            $result = $this->adsmedia->ping();
            return response()->json(['success' => true, 'data' => $result]);
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }
}

